core/primitives/src/network.rs

```diff
@@ -20,8 +20,8 @@ impl PeerId {
         Self(key)
     }
 
-    pub fn public_key(&self) -> PublicKey {
-        self.0.clone()
+    pub fn public_key(&self) -> &PublicKey {
```

Comment:
Feels a bit strange that `peer_id.0` and `public_key()` are both public synonyms for the same thing, I'd probably expect the field to be private and to have `From<PeerId> for PublicKey` if we need it.